src/Components/Patient/PatientRegister.tsx

```diff
@@ -665,10 +675,25 @@ export const PatientRegister = (props: PatientRegisterProps) => {
       const medData = MEDICAL_HISTORY_CHOICES.find((i) => i.id === id);
       if (medData) {
         const details = formData[`medical_history_${medData.id}`];
-        medical_history.push({
-          disease: medData.text,
-          details: details ? details : "",
-        });
+        if (medData.id === 12) {
+          medical_history.push({
+            disease: medData.text,
+            details: details ? details : "",
+            status: formData["tb_status"] ?? undefined,
+            duration: Number(formData["tb_duration"]) ?? undefined,
+          });
+        } else if (medData.id === 14) {
+          medical_history.push({
+            disease: medData.text,
+            details: details ? details : "",
+            type: formData["cancer_type"] ?? undefined,
+          });
+        } else {
+          medical_history.push({
+            disease: medData.text,
+            details: details ? details : "",
+          });
+        }
```

Comment:
1. Why explicitly fallback to undefined when the value maybe already undefined?
2. Add comments for 12 and 14 disease codes so that it is easily readable and understandable by an external contributor.

```suggestion
        if (medData.id === 12) {
          // Add comment for which disease 12 is
          medical_history.push({
            disease: medData.text,
            details: details || "",
            status: formData["tb_status"],
            duration: Number(formData["tb_duration"]),
          });
        } else if (medData.id === 14) {
          // Add comment for which disease 14 is
          medical_history.push({
            disease: medData.text,
            details: details || "",
            type: formData["cancer_type"],
          });
        } else {
          medical_history.push({
            disease: medData.text,
            details: details || "",
          });
        }
```